rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -74,27 +74,57 @@ impl OpQueue {
         // The other consideration is whether to put the channel receiver in the OpQueue or in a dedicated task
         // that also holds an Arc to the Mutex. For simplicity, we'll put it in the OpQueue for now.
         let mut message_retry_requests = vec![];
-        while let Ok(message_id) = self.retry_rx.lock().await.try_recv() {
-            message_retry_requests.push(message_id);
+
+        while let Ok(retry_request) = self.retry_receiver.lock().await.try_recv() {
+            let uuid = retry_request.uuid.clone();
+            message_retry_requests.push((
+                retry_request,
+                MessageRetryResponse {
+                    uuid,
+                    evaluated: 0,
+                    matched: 0,
+                },
+            ));
         }
+
         if message_retry_requests.is_empty() {
             return;
         }
+
         let mut queue = self.queue.lock().await;
+        let queue_length = queue.len();
+
         let mut reprioritized_queue: BinaryHeap<_> = queue
             .drain()
             .map(|Reverse(mut op)| {
-                if message_retry_requests.iter().any(|r| r.op_matches(&op)) {
+                let matches = message_retry_requests
+                    .iter_mut()
+                    .map(|(retry_req, req_metric)| {
+                        let match_res = retry_req.pattern.op_matches(&op);
+                        // update retry metrics
+                        if match_res {
+                            req_metric.matched += 1
+                        }
+                        match_res
+                    })
+                    .any(|x| x);
+                if matches {
                     info!(
```

Comment:
realizing it'd be useful to include the retry request in this log

Reply:
we can't really log the specific retry request because we only check if some retry request matches the operation.
Unless we want to log each retry request that matches in the `.map()`

Reply:
you're right, though we could take advanced of the `filter` result and accumulate it into a vec of maps, so we include all request IDs that matched in the log. It'd also be cleaner to refactor everything inside the `.map(|Reverse(mut op)| {` into its own function, since it's getting quite crowded